src/strategies/base.ts

```diff
@@ -428,6 +428,30 @@ export abstract class BaseStrategy implements Strategy {
               );
           }
         }
+      } else if (extraFile.endsWith('.json')) {
+        extraFileUpdates.push({
+          path: this.addPath(extraFile),
+          createIfMissing: false,
+          updater: new GenericJson('$.version', version),
+        });
+      } else if (extraFile.endsWith('.yaml') || extraFile.endsWith('.yml')) {
+        extraFileUpdates.push({
+          path: this.addPath(extraFile),
+          createIfMissing: false,
+          updater: new GenericYaml('$.version', version),
+        });
+      } else if (extraFile.endsWith('.toml')) {
+        extraFileUpdates.push({
+          path: this.addPath(extraFile),
+          createIfMissing: false,
+          updater: new GenericToml('$.version', version),
+        });
+      } else if (extraFile.endsWith('.xml')) {
+        extraFileUpdates.push({
+          path: this.addPath(extraFile),
+          createIfMissing: false,
+          updater: new GenericXml('/*/version', version),
```

Comment:
Additionally, we might want to have a confidence test to ensure that this xpath only updates the first-level `version` field.

i.e.

```xml
<some-root>
  <version>1.2.3</version><!-- this one should be updated -->
  <blah>
    <version>2.3.4</version><!-- this one should NOT be updated -->
  </blah>
</some-root>
```

Reply:
Or if it's basic knowledge of xpath, then add a comment describing what this is matching.

Reply:
[`/*/version`](http://xpather.com/f7pP8ml2):

- `/` to start the path to the root of the XML document.
- `*` to pass the root element (without filtering on its name).
- `/` to select all directed children.
- `version` to select elements named `version`.

I added a comment.